Assignment 2.Rmd

```diff
@@ -91,10 +91,61 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.widt
 
 In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
 ```{r}
-plot(g,layout=layout.fruchterman.reingold,vertex.size=EDGE$count*5,)
+library(tidyr)
+library(dplyr)
+library(igraph)
+student_sum<-EDGE%>%group_by(to)%>%summarise(sum(count))
+names(student_sum)<-c("id","count")
+teacher_sum<-EDGE%>%group_by(from)%>%summarise(sum(count))
+names(teacher_sum)<-c("id","count")
+vertex1<-bind_rows(teacher_sum,student_sum)
+g<-graph.data.frame(EDGE,directed = TRUE,vertices = vertex1)
+plot(g,layout=layout.fruchterman.reingold,edge.length=EDGE$count*10,edge.arrow.size=0.7,vertex.size=vertex1$count*2)
 ```
+```{r}
+library(tidyr)
+library(dbplyr)
+Df<-hudk4050.classes
+D3<-unite(Df,V1,1:2)
+names(D3)<-c("Name","Class_1","Class_2","Class_3","Class_4","Class_5","Class_6")
+D3<-D3[-1,]
+D4<-gather(D3,Name,Class)
+D4[,1]<-D3$Name
+D4$Class<-gsub(pattern = "\\s",replacement = "",D4$Class)
+cnt<-ifelse(D4$Class=="",D4$count<-0,D4$count<-1)
+D4$count<-cnt
+classes<-unique(D4$Class)
+D4[34,2]<-"HUDK4050"
+D4[87,2]<-"HUDM4125"
+D4[140,2]<-"HUDM5026"
+D4[193,2]<-"HUDM5126"
+D5<-D4[!D4$Class=="",]
```

Comment:
Nice cleaning. Maybe you can directly remove rows with class=="" without adding 0.